src/org/myworldgis/netlogo/ProjectLatLon.java

```diff
@@ -0,0 +1,127 @@
+        }
+    }
+
+    /**
+     * 
+     */
+    public static final strictfp class ProjectWGS84 extends GISExtension.Reporter {
+
+        /** */
+        public String getAgentClassString() {
+            return "OTPL";
+        }
+        
+        /** */
+        public Syntax getSyntax() {
+            return SyntaxJ.reporterSyntax(new int[] { Syntax.NumberType(), Syntax.NumberType() },
+                                        Syntax.ListType());
+        }
+        
+        /** */
+        public Object reportInternal (Argument args[], Context context) 
+                throws ExtensionException , LogoException {
+            double lat = args[0].getDoubleValue();
+            double lon = args[1].getDoubleValue();
+            return projectPointGivenEllipsoid(lat, lon, WGS84);
+        }
+    }
+
+    /**
+     * 
+     */
+    public static Object projectPointGivenEllipsoid(double lat, double lon, Ellipsoid srcEllipsoid)
+            throws ExtensionException , LogoException {
+        LogoListBuilder result = new LogoListBuilder();
+        if(lat > 90 || lat < -90 || lon > 180 || lon < -180){
+            return result.toLogoList();
+        }
+        Projection dstProj = GISExtension.getState().getProjection();
+        if (dstProj == null){
+            throw new ExtensionException("You must use gis:load-coordinate-system or gis:set-coordinate-system before you can project lat/lon pairs.");
+        }
+
+        Geometry point = GISExtension.getState().factory().createPoint(new Coordinate(lon, lat));
+        if (point == null){
+            return result.toLogoList();
+        }
+        
+        Ellipsoid dstEllipsoid = dstProj.getEllipsoid();
+        // in cases where the current projection uses a geographic (as opposed to projected)
+        // coordinate system (i.e. values are expressed in terms of angular distance as 
+        // opposed to linear distance on a 2D projection) and the given ellipsoid
+        // is the same one used as the destination projection, do not reproject and 
+        // introduce precision errors. cf. similar behavior in LoadDataset.java
+        boolean shouldReproject = !(dstProj instanceof Geographic) || !srcEllipsoid.equals(dstEllipsoid);
+        if(shouldReproject){
+            GeometryTransformer forward = dstProj.getForwardTransformer();
+            GeometryTransformer inverse = new Geographic(srcEllipsoid, Projection.DEFAULT_CENTER, NonSI.DEGREE_ANGLE).getInverseTransformer();
+            point = forward.transform(inverse.transform(point));
+        }
+
+        Coordinate projected = point.getCoordinate();
+        if (projected == null){
+            return result.toLogoList();
+        }
+
+        Coordinate transformed = GISExtension.getState().gisToNetLogo(projected, null);
+        if(transformed != null){
+            result.add(Double.valueOf(transformed.x));
+            result.add(Double.valueOf(transformed.y));
+        }
+        return result.toLogoList();
+    }
+}
```

Comment:
Make sure to leave a blank line at the end of the file (most IDEs have an option to add/maintain it).